docs/data/joy/customization/right-to-left/right-to-left.md

```diff
@@ -0,0 +1,127 @@
+# Right-to-left language support
+
+<p class="description">Learn how to implement right-to-left (RTL) text with Material UI to support languages such as Arabic, Persian, and Hebrew.</p>
+
+This guide outlines the three steps necessary to change the direction of text-based components in Joy UI to support RTL languages, as shown in the demo below:
+
+<!-- {{"demo": "Direction.js"}} -->
+
+## 1. Set the HTML direction
+
+### Globally
+
+Add `dir="rtl"` to the app's root `<html>` tag to set the global text direction:
+
+```html
+<html dir="rtl"></html>
+```
+
+If your React app doesn't control the root `<html>` and you need to change the text direction at runtime, use the JavaScript API instead:
+
+```js
+document.dir = 'rtl';
+```
+
+### Locally
+
+Add the `dir="rtl"` attribute to any other HTML element or React component if you need limit the scope of the text direction to that element and its children.
+
+:::warning
+This won't work with portaled elements, such as Dialogs, as they render outside of the element with the `dir` attribute.
+
+To fix it, make sure to add the `dir` attribute directly to them:
```

Comment:
I don't totally understand what this is saying, and I'm having a hard time wrapping my head around it. Is it saying that you can't add the `dir="rtl"` to a child of a component that makes use of React portals? So if you want that child to be RTL, you need to apply the attribute to that parent component instead?

Reply:
I think I know what this is about — say you have a container with a button inside it that opens a dialog:

```
<div dir="rtl">
  <button/>
</div>
<dialog dir="rtl" />
```

The button would be affected by the parent div having the `rtl` attribute. But, given the opened dialog would be _outside_ the div in the DOM tree, it wouldn't inherit its `rtl`. This is why we'd also need to add it to the dialog explicitly. Does that make sense?

That's different from the above-mentioned global approach, as setting the attribute directly on the `html` tag would affect any other element down the tree.